src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -104,29 +104,12 @@ public class MempoolBehavior : NetworkPeerBehavior
         /// <summary>Time of last memory pool request in unix time.</summary>
         public long LastMempoolReq { get; private set; }
 
-        /// <summary>Time of next inventory send in unix time.</summary>
-        public long NextInvSend { get; set; }
-
         /// <summary>Whether memory pool is in state where it is ready to send it's inventory.</summary>
         public bool CanSend
         {
             get
             {
-                if (!this.AttachedPeer?.PeerVersion?.Relay ?? true)
-                    return false;
-
-                // Check whether periodic sends should happen
-                bool sendTrickle = this.AttachedPeer.Behavior<ConnectionManagerBehavior>().Whitelisted;
-
-                if (this.NextInvSend < this.mempoolManager.DateTimeProvider.GetTime())
-                {
-                    sendTrickle = true;
-                    // Use half the delay for outbound peers, as there is less privacy concern for them.
-                    this.NextInvSend = this.mempoolManager.DateTimeProvider.GetTime() + 10;
-                    // TODO: PoissonNextSend(nNow, InventoryBroadcastInterval >> !pto->fInbound);
-                }
-
-                return sendTrickle;
+                return (this.AttachedPeer?.PeerVersion?.Relay ?? false);
```

Comment:
Maybe remove the property completely? 

CanSend is called from 2 places- when we call SendTrickleAsync - here just modify the linq expression to avoid selecting peers with CanRelay ==false, and in 2nd place just remove it since it can't happen (and even if CanRelay==false peer asks us for our mempool we don't mind sending it).

Reply:
I am fine with that but we still need to honor `Relay` somehow.
And it might be we need to also add this flag in mempool reply code

Reply:
I looked at the usages of CanSend and basically it's needed in one place- when we send trickle (and there we have peer in our hands so we can select peers with Relay == true).

In the `SendMempoolPayloadAsync` we don't really need it. Actually quite the opposite- we are disallowing light wallets to be able to ask if their tx was included in the mempool (currently we don't do that because of the privacy issues but I can see why someone would like to do that).

Reply:
yeah we can indeed filter at the trickle method.